File: app/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/activities/fragments/titlescreen/applanguages/views/LanguagesAdapterView.java
package com.TritiumGaming.phasmophobiaevidencepicker.activities.fragments.titlescreen.applanguages.
        views;

import android.content.Context;
import android.os.Build;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.appcompat.widget.AppCompatImageView;
import androidx.appcompat.widget.AppCompatTextView;
import androidx.recyclerview.widget.RecyclerView;

import com.TritiumGaming.phasmophobiaevidencepicker.R;

import java.util.ArrayList;

public class LanguagesAdapterView extends RecyclerView.Adapter<LanguagesAdapterView.ViewHolder> {
    private static int mPreviousIndex = 0;
    private final ArrayList<String> languages;
    private final OnLanguageListener onLanguageListener;

    public LanguagesAdapterView(
            ArrayList<String> languages,
            int selected,
            OnLanguageListener onLanguageListener) {
        mPreviousIndex = selected;
        this.languages = languages;
        this.onLanguageListener = onLanguageListener;
    }

    public static class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        public AppCompatTextView label_languageTitle;
        public AppCompatImageView image;
        private final OnLanguageListener onLanguageListener;

        public ViewHolder(View view, OnLanguageListener onLanguageListener) {
            super(view);
            label_languageTitle = itemView.findViewById(R.id.textView_languageName);
            image = itemView.findViewById(R.id.imageView_languageChoiceIcon);
            view.setOnClickListener(this);
            this.onLanguageListener = onLanguageListener;
        }

        @Override
        public void onClick(View v) {
            this.onLanguageListener.onNoteClick(getAdapterPosition());
            mPreviousIndex = getAdapterPosition();
        }
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        Context context = parent.getContext();
        LayoutInflater inflater = LayoutInflater.from(context);
        View messageView = inflater.inflate(R.layout.item_language, parent, false);
        return new LanguagesAdapterView.ViewHolder(messageView, this.onLanguageListener);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        AppCompatTextView textView = holder.label_languageTitle;
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            textView.setAutoSizeTextTypeUniformWithConfiguration(
                    12,
                    30,
                    1,
                    AppCompatTextView.AUTO_SIZE_TEXT_TYPE_UNIFORM);
        } else {
            textView.setAutoSizeTextTypeUniformWithConfiguration(
                    12,
                    30,
                    1,
                    1);
        }
        textView.setText(languages.get(position));

        //color on item unselecting item
        holder.image.setVisibility(mPreviousIndex == position ? View.VISIBLE: View.INVISIBLE);
    }

    @Override
    public int getItemCount() {
        return languages.size();
    }

    public interface OnLanguageListener {
        void onNoteClick(int position);
    }

}

File: app-view/src/main/java/com/tritiumgaming/phasmophobiaevidencepicker/presentation/ui/activities/mainmenus/applanguages/views/LanguagesAdapterView.kt
package com.tritiumgaming.phasmophobiaevidencepicker.presentation.ui.activities.mainmenus.applanguages.views

import android.graphics.drawable.Drawable
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.widget.AppCompatTextView
import androidx.recyclerview.widget.RecyclerView
import com.tritiumgaming.phasmophobiaevidencepicker.R
import com.tritiumgaming.phasmophobiaevidencepicker.presentation.viewmodel.GlobalPreferencesViewModel
import com.tritiumgaming.phasmophobiaevidencepicker.presentation.ui.views.global.PETImageButton

class LanguagesAdapterView(
    languages: ArrayList<GlobalPreferencesViewModel.LanguageObject>, selected: Int,
    onLanguageListener: OnLanguageListener
) : RecyclerView.Adapter<LanguagesAdapterView.ViewHolder>() {
    private val languages: ArrayList<GlobalPreferencesViewModel.LanguageObject>
    private val onLanguageListener: OnLanguageListener

    companion object {
        private var mPreviousIndex = 0
    }

    init {
        mPreviousIndex = selected
        this.languages = languages
        this.onLanguageListener = onLanguageListener
    }

    class ViewHolder(view: View, onLanguageListener: OnLanguageListener) :
        RecyclerView.ViewHolder(view), View.OnClickListener {
        val nameTextView: AppCompatTextView? = itemView.findViewById(R.id.textView_languageName)
        val image: PETImageButton? = itemView.findViewById(R.id.imageView_languageChoiceIcon)
        val background: Drawable? = nameTextView?.background
        private val onLanguageListener: OnLanguageListener

        init {
            view.setOnClickListener(this)
            this.onLanguageListener = onLanguageListener
        }

        override fun onClick(v: View) {
            onLanguageListener.onNoteClick(adapterPosition)
            mPreviousIndex = adapterPosition
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val context = parent.context
        val inflater = LayoutInflater.from(context)
        val messageView = inflater.inflate(R.layout.item_language, parent, false)
        return ViewHolder(messageView, this.onLanguageListener)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.nameTextView?.let { textView ->
            textView.text = languages[position].name
            textView.isSelected = true
        }
        //color on item unselecting item
        holder.image?.visibility = if (mPreviousIndex == position) View.VISIBLE else View.INVISIBLE
        holder.background?.setLevel(if (mPreviousIndex == position) 1 else 0)
    }

    override fun getItemCount(): Int {
        return languages.size
    }

    interface OnLanguageListener {
        fun onNoteClick(position: Int)
    }

}